platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/WiringModel.java

```diff
@@ -0,0 +1,129 @@
+
+/**
+ * A wiring model is a collection of task schedulers and the wires connecting them. It can be used to analyze the wiring
+ * of a system and to generate diagrams.
+ */
+public abstract class WiringModel {
+
+    private final PlatformContext platformContext;
+    private final Time time;
+
+    /**
+     * Constructor.
+     *
+     * @param platformContext the platform context
+     * @param time            provides wall clock time
+     */
+    protected WiringModel(@NonNull final PlatformContext platformContext, @NonNull final Time time) {
+        this.platformContext = Objects.requireNonNull(platformContext);
+        this.time = Objects.requireNonNull(time);
+    }
+
+    /**
+     * Build a new wiring model instance.
+     *
+     * @param platformContext the platform context
+     * @param time            provides wall clock time
+     * @return a new wiring model instance
+     */
+    @NonNull
+    public static WiringModel create(@NonNull final PlatformContext platformContext, @NonNull final Time time) {
+        return new StandardWiringModel(platformContext, time);
+    }
+
+    /**
+     * Get a new task scheduler builder.
+     *
+     * @param name the name of the task scheduler. Used for metrics and debugging. Must be unique. Must only contain
+     *             alphanumeric characters and underscores.
+     * @return a new wire builder
+     */
+    @NonNull
+    public final <O> TaskSchedulerBuilder<O> schedulerBuilder(@NonNull final String name) {
+        return new TaskSchedulerBuilder<>(this, name);
+    }
+
+    /**
+     * Get a new task scheduler metrics builder. Can be passed to
+     * {@link TaskSchedulerBuilder#withMetricsBuilder(TaskSchedulerMetricsBuilder)} to add metrics to the task
+     * scheduler.
+     *
+     * @return a new task scheduler metrics builder
+     */
+    @NonNull
+    public final TaskSchedulerMetricsBuilder metricsBuilder() {
+        return new TaskSchedulerMetricsBuilder(platformContext.getMetrics(), time);
+    }
+
+    /**
+     * Check to see if there is cyclic backpressure in the wiring model. Cyclical back pressure can lead to deadlocks,
+     * and so it should be avoided at all costs.
+     *
+     * <p>
+     * If this method finds cyclical backpressure, it will log a message that will fail standard platform tests.
+     *
+     * @return true if there is cyclical backpressure, false otherwise
+     */
+    public abstract boolean checkForCyclicalBackpressure();
+
+    /**
+     * Generate a mermaid style wiring diagram.
+     *
+     * @param groups optional groupings of vertices
+     * @return a mermaid style wiring diagram
+     */
+    @NonNull
+    public abstract String generateWiringDiagram(@NonNull final Set<ModelGroup> groups);
+
+    /**
+     * Reserved for internal framework use. Do not call this method directly.
+     * <p>
+     * Register a vertex in the wiring model. These are either output task schedulers or wire transformers. Vertexes
```

Comment:
The plural of "vertex" is "vertices". There are couple instances of "vertexes"

Reply:
fixed